### PR TITLE
Feature/add missing events to nostra transform events

### LINKED ISSUE #338: [DataHandler] Add missing events to nostra transform_events
1. Before applying, Read our [Contributor Guidelines](https://github.com/CarmineOptions/derisk-research/blob/master/CONTRIBUTING.md) and [README](https://github.com/CarmineOptions/derisk-research/blob/master/README.md). 
2. In this [file](https://github.com/CarmineOptions/derisk-research/blob/master/apps/data_handler/handlers/events/nostra/transform_events.py), add next missing events from this [file](https://github.com/CarmineOptions/derisk-research/blob/master/apps/data_handler/handler_tools/data_parser/serializers/nostra.py)
- `parse_interest_bearing_collateral_mint_event`
- `parse_non_interest_bearing_collateral_burn_event`
- `parse_non_interest_bearing_collateral_mint_event`
- `parse_interest_rate_model_event`
3. Event information you can find [here](https://github.com/CarmineOptions/derisk-research/blob/master/apps/data_handler/handlers/loan_states/nostra_alpha/events.py#L161)
4. DB models to save you can find [here](https://github.com/CarmineOptions/derisk-research/blob/master/apps/data_handler/db/models/nostra_events.py) (if some of models are missing, please add them, don't forget to add also migration file)
5. Add test cases to your code [here](https://github.com/CarmineOptions/derisk-research/blob/master/apps/data_handler/tests/test_nostra_transformer.py)

### PR DESCRIPTION
**Pull request by MrRoudyk**
Closes #338

### CODE REVIEW

File: apps/data_handler/handler_tools/data_parser/serializers/nostra.py [hunk 1]
```diff
@@ -34,6 +35,15 @@ def validate_address(cls, value: str, info: ValidationInfo) -> str:
             raise ValueError(f"Invalid address provided for {info.field_name}")
         return add_leading_zeros(value)
 
+    @field_validator("amount", mode="before")
+    def validate_numeric_string(cls, value: str, info: ValidationInfo) -> Decimal:
```
Comment (djeck1432) [inline apps/data_handler/handler_tools/data_parser/serializers/nostra.py:39]: Please, add docstring

File: apps/data_handler/handler_tools/data_parser/serializers/nostra.py [hunk 2]
```diff
@@ -66,6 +76,15 @@ def validate_address(cls, value: str, info: ValidationInfo) -> str:
...
```
Comment (djeck1432) [inline apps/data_handler/handler_tools/data_parser/serializers/nostra.py:80]: Add docstring

File: apps/data_handler/db/crud.py [hunk 3]
```diff
@@ -942,15 +997,41 @@ def create_debt_transfer_event(
...
+        finally:
+            db.close()
+
+    def create_non_interest_bearing_collateral_burn_event(
+        self, protocol_id: str, event_name: str, block_number: int, event_data: dict
+    ) -> None:
+        """
+        Creates a NonInterestBearingCollateralBurnEventModel record in the database.
+        """
+        db = self.Session()
```
Comment (djeck1432) [inline apps/data_handler/db/crud.py:1032]: why not use write_to_db method?
please change it in another places as well

**Review by djeck1432 — CHANGES_REQUESTED** [review]
1. I don't understand why did you rename models name and tables
2. Please run also generating migration file

**Comment by MrRoudyk** [comment]
> 1. I don't understand why did you rename models name and tables
But I didn't rename models name and tables. I just added new methods and sorted them alphabetically

**Comment by djeck1432** [comment]
@MrRoudyk Maybe you resolved conflict in wrong way, let me check

**Comment by djeck1432** [comment]
@MrRoudyk okay, it's just github display in a wrong way, checked all looks good regarding models

File: apps/data_handler/db/crud.py (hunk 3 above)
Comment (MrRoudyk) [reply apps/data_handler/db/crud.py]: I refactored the code to use the `write_to_db` method and added a new method, `get_all_events_from_models`, to the base `DBConnector` class

**Comment by MrRoudyk** [comment]
> 2. Please run also generating migration file

I’m still working on generating the migration file following the instructions but haven’t succeeded yet

**Comment by MrRoudyk** [comment]
> > 2. Please run also generating migration file
> 
> I’m still working on generating the migration file following the instructions but haven’t succeeded yet

Oh, I realized that since there weren't any actual changes to the models or table names, we don't need to create a migration file.
@djeck1432 let me know if that works for you or if there's anything else I should do

**Comment by djeck1432** [comment]
@MrRoudyk https://github.com/CarmineOptions/derisk-research/pull/349/files
try to run this script for migration

**Comment by djeck1432** [comment]
@MrRoudyk Just checked, looks good

**Review by djeck1432 — APPROVED** [review]
Looks good to me

**Comment by MrRoudyk** [comment]
> @MrRoudyk https://github.com/CarmineOptions/derisk-research/pull/349/files try to run this script for migration

Yes, I ran the script and it worked (I just had to replace `poetry shell` with `poetry run`). The migration file is here: https://github.com/MrRoudyk/derisk-research/blob/feature/add-missing-events-to-nostra-transform_events/apps/data_handler/migrations/versions/cc5f9168f89a_add_missing_events_to_nostra_transform_.py

**Pull request merged**